pkg/instance/service.go

```diff
@@ -313,6 +315,63 @@ func (s service) deployDeploymentInstance(token string, instance *model.Deployme
 	return nil
 }
 
+func (s service) DeleteDeployment(deployment *model.Deployment) error {
+	deploymentGraph, err := s.validateNoCycles(deployment.Instances)
+	if err != nil {
+		return err
+	}
+
+	instances, err := deploymentOrder(deployment, deploymentGraph)
+	if err != nil {
+		return err
+	}
+	slices.Reverse(instances)
+
+	var errs error
+	for _, instance := range instances {
+		err := s.destroyDeploymentInstance(instance)
+		if err != nil {
+			errs = errors.Join(errs, fmt.Errorf("failed to destroy instance(%s) %q: %v", instance.StackName, instance.Name, err))
+		}
+	}
+
+	if errs != nil {
```

Comment:
What if we succeeded in destroying only some instances, should we delete these from the DB? 🤔 Imagine that case and a user retrying, would destroying the same instances that were destroyed before work again? Hence can a user retry and actually get all instances destroyed and removed from the DB in that case.